src/mock.js

```diff
@@ -82,6 +82,17 @@ function addMockFunctionsToSchema({ schema, mocks = {}, preserveResolvers = fals
     return customMock;
   }
 
+  function assignResolveType(type) {
+    const fieldType = getNullableType(type);
+    const namedFieldType = getNamedType(fieldType);
+
+    if (namedFieldType instanceof GraphQLUnionType ||
+        namedFieldType instanceof GraphQLInterfaceType
+    ) {
+      namedFieldType.resolveType = (data, context, info) => info.schema.getType(data.typename);
```

Comment:
I think we should make sure not to overwrite `resolveType` if it's provided, or at least make it an option to preserve existing ones (there's the `preserveResolvers` option, which you could co-opt for it).